Sources/AsyncHTTPClient/ConnectionPool/ChannelHandler/HTTP1ProxyConnectHandler.swift

```diff
@@ -155,7 +160,13 @@ final class HTTP1ProxyConnectHandler: ChannelDuplexHandler, RemovableChannelHand
             method: .CONNECT,
             uri: "\(self.targetHost):\(self.targetPort)"
         )
-        head.headers.replaceOrAdd(name: "host", value: "\(self.targetHost)")
+        if !head.headers.contains(name: "host") {
```

Comment:
The `HTTPRequestHead` is created in the lines above and we can therefore say for sure we will never have a `host` set. This makes sense because we don't need to look into the headers we have created here but from the original HTTP request from the user. This will require plumping the `HTTPHeader`s from the original request through the stack. 

FYI: @rnro is currently working on moving `HTTP1ProxyConnectHandler` to `swift-nio-extras` [in this PR](https://github.com/apple/swift-nio-extras/pull/185). In this new version the `HTTPHeader`s are passed into the `HTTP1ProxyConnectHandler.init` which makes this a bit easier. We still need to get the original request headers to where we create the `HTTP1ProxyConnectHandler` in AHC.